squeal-postgresql-ltree/src/Squeal/PostgreSQL/LTree.hs

```diff
@@ -142,10 +144,16 @@ newtype LTree = UnsafeLTree {getLTree :: Text}
   deriving stock (Eq,Ord,Show,Read,Generic)
 -- | `PGltree`
 instance IsPG LTree where type PG LTree = PGltree
-instance FromPG LTree where
-  fromPG = UnsafeLTree <$> devalue Decoding.text_strict
-instance ToPG db LTree where
-  toPG = pure . Encoding.text_strict . getLTree
+instance TypeError
+  ('Text "PostgreSQL does not support binary output for ltree, cast it to text.")
```

Comment:
PostgreSQL 13 supports it!
https://www.postgresql.org/docs/13/release-13.html
> Add support for binary I/O of ltree, lquery, and ltxtquery types (Nino Floris)

Reply:
Cool. Do you know where the code for those binary I/O functions is?

Reply:
[Here](https://github.com/npgsql/npgsql/pull/2991/files#diff-2f8c3cd6f8f591b5810ccd0731289bfb754bb8078aed3ea62ffca89dbfb3dae0R23) it is.